src/main/java/net/dv8tion/jda/api/utils/messages/MessageCreateRequest.java

```diff
@@ -325,11 +325,18 @@ default R addFiles(@Nonnull FileUpload... files)
     default R applyData(@Nonnull MessageCreateData data)
     {
         Checks.notNull(data, "MessageCreateData");
+
+        final List<LayoutComponent> layoutComponents = data.getComponents().stream()
+                .map(LayoutComponent::createCopy)
+                .collect(Collectors.toList());
+        final List<FileUpload> files = data.getFiles().stream()
+                .map(f -> FileUpload.fromData(f.getData(), f.getName()).setDescription(f.getDescription()))
```

Comment:
I don't want file uploads to be copied this way because it opens up for issues with concurrent reads to the same input stream.

Reply:
How then ? passing existing file uploads ?

Reply:
I think we just have to document that files do not get copied over as a limitation.
If people want to introduce concurrent problems by sharing the same FileUpload, that's one them.

It'd also be good to not try to copy files over anyways because you wouldn't be able to copy over attachments as they'd only be valid if you were editing a the same message.